apps/test/unit/JSInterpreterTest.js

```diff
@@ -274,6 +274,312 @@ myCallback("this message is coming from inside the interpreter");
 
   });
 
+  let aceEditor;
+  function setupFakeAce() {
+    let oldAce;
+    beforeEach(() => {
+      oldAce = window.ace;
+      window.ace = {
+        require: sinon.stub().returns({Range: sinon.stub()}),
+      };
+      const breakpoints = [];
+      const aceSession = {
+        addMarker: sinon.spy(),
+        getBreakpoints: sinon.stub().returns(breakpoints),
+      };
+      aceEditor = {
+        isRowFullyVisible: () => true,
+        getSession: () => aceSession,
+        getSelection: () => ({
+          getRange: () => ({
+            start: {},
+            end: {},
+          }),
+          clearSelection: () => {},
+        })
+      };
+    });
+    afterEach(() => {
+      window.ace = oldAce;
+    });
+  }
```

Comment:
I know it's only here for now, but it'd be nice to extract this to a utility file.  It's even the sort of thing I see us eventually publishing as its own little package, generally useful as it grows and gets more flexible.  For now I'd imagine something that follows sinon's patterns, like this:

```javascript
import fakeAce from './fakeAce';

function setupFakeAce() {
  beforeEach(() => fakeAce(window));
  afterEach(() => window.ace.restore());
}
```